File: app/src/main/java/com/blues/module_gank/old/model_fuli/view/FuliViewHolder.java
package com.blues.module_gank.old.model_fuli.view;

import android.app.Activity;
import android.content.Intent;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.blues.R;
import com.blues.constant.UsedConstant;
import com.blues.framework.utils.ScreenUtil;
import com.blues.module_gank.old.ShowImageActivity;
import com.blues.module_gank.old.model_fuli.model.FuliEntity;
import com.bumptech.glide.Glide;
import com.jude.easyrecyclerview.adapter.BaseViewHolder;

import androidx.core.app.ActivityOptionsCompat;

public class FuliViewHolder extends BaseViewHolder<FuliEntity.ResultsBean> {

    private ImageView ivAvatar, ivUrl;
    private TextView tvDesc, tvAuthor, tvSource, tvPTime;

    public FuliViewHolder(ViewGroup parent) {
        super(parent, R.layout.item_fuli);

        tvDesc = $(R.id.tv_desc);
        tvAuthor = $(R.id.tv_author);
        tvSource = $(R.id.tv_source);
        tvPTime = $(R.id.tv_publish_time);
        ivAvatar = $(R.id.iv_avatar);
        ivUrl = $(R.id.iv_url);
    }

    @Override
    public void setData(FuliEntity.ResultsBean data) {
        super.setData(data);

//        tvDesc.setText(data.getDesc());
        ivUrl.setLayoutParams(new LinearLayout.LayoutParams(ScreenUtil.getScreenWidth() * 2 / 3, ScreenUtil.getScreenHeight() / 3));
        Glide.with(ivUrl.getContext()).load(data.getUrl()).placeholder(R.mipmap.ic_img_error).into(ivUrl);
        tvAuthor.setText(String.format("作者：%s", data.getWho()));
        tvSource.setText(String.format("来自：%s", data.getSource()));
        tvPTime.setText(String.format("日期：%s", data.getPublishedAt().substring(0, data.getPublishedAt().indexOf("T"))));

        ivUrl.setOnClickListener(v -> {
            Intent intent = new Intent(getContext(), ShowImageActivity.class);
            intent.putExtra(UsedConstant.IMAGE_URL, data.getUrl());
            ActivityOptionsCompat options = ActivityOptionsCompat.makeSceneTransitionAnimation((Activity) getContext(), ivUrl, "transitionImage");
            getContext().startActivity(intent, options.toBundle());
        });
    }
}

File: app/src/main/java/com/blues/gankio/v1/view/FuliViewHolder.kt
package com.blues.gankio.v1.view

import android.app.Activity
import android.content.Intent
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.core.app.ActivityOptionsCompat
import com.blues.R
import com.blues.constant.IMAGE_URL
import com.blues.framework.utils.ScreenUtil.screenHeight
import com.blues.framework.utils.ScreenUtil.screenWidth
import com.blues.gankio.v1.ShowImageActivity
import com.blues.gankio.v1.model.GankBean
import com.bumptech.glide.Glide
import com.jude.easyrecyclerview.adapter.BaseViewHolder

class FuliViewHolder(parent: ViewGroup?) :
    BaseViewHolder<GankBean.ResultsBean>(parent, R.layout.item_fuli) {

    private val ivAvatar: ImageView = itemView.findViewById(R.id.tv_author)
    private val ivUrl: ImageView = itemView.findViewById(R.id.iv_url)
    private val tvDesc: TextView = itemView.findViewById(R.id.tv_desc)
    private val tvAuthor: TextView = itemView.findViewById(R.id.iv_avatar)
    private val tvSource: TextView = itemView.findViewById(R.id.tv_source)
    private val tvPTime: TextView = itemView.findViewById(R.id.tv_publish_time)

    override fun setData(data: GankBean.ResultsBean) {
        super.setData(data)

        //        tvDesc.setText(data.getDesc());
        ivUrl.layoutParams = LinearLayout.LayoutParams(screenWidth * 2 / 3, screenHeight / 3)
        Glide.with(ivUrl.context).load(data.url).placeholder(R.mipmap.ic_img_error).into(ivUrl)
        tvAuthor.text = "作者：${data.who}"
        tvSource.text = "来自：${data.source}"
        tvPTime.text = "日期：${data.publishedAt.substring(0, data.publishedAt.indexOf("T"))}"
        ivUrl.setOnClickListener {
            val intent = Intent(context, ShowImageActivity::class.java).putExtra(IMAGE_URL, data.url)
            val options = ActivityOptionsCompat.makeSceneTransitionAnimation((context as Activity), ivUrl, "transitionImage")
            context.startActivity(intent, options.toBundle())
        }
    }
}